### DISCUSSION TITLE
Different volumeSizeLimitMB depending on the volume server

### DISCUSSION THREAD

**Discussion by wokalski** [seaweedfs/seaweedfs#6147] [Q&A] [answered]
Hey, I am configuring seaweedfs and I have very different disks:
1. HDDs with 22TB of storage
2. NVMEs with 960 GB of storage
3. NVMEs with 128 GB of storage

Since volumeSizeLimitMB is set on the master level, all of my volume severs will use the same volume size. My understanding is that it's suboptimal. 30GB is perhaps a good default for the 960 GB NVMe but probably not a good one for the large HDD or small NVMe. What's the recommended setup in such situation? My primary purpose is running seaweed backed S3 compatible object storage.

**Comment by chrislusf** [comment]
* The smaller NVME disks could be very good for volume indexes.
* Imagine each volume is a virtual portable disk, which can move to different actual disks. If the volume size is optimal for one particular disk type, it is less optimal for other disk type. Different volume size would make tiering much more complicated. Maybe a good direction for future versions.

**Reply by wokalski** [reply]
We already have this kind of distinction with the disk types. Provided we don't support mixing large disk and regular disk volumes, could we move this volume size setting to the volume server?

**Reply by chrislusf** [reply, marked as answer]
* volumes can be moved to different disk types. With different sizes, this would need to be disabled.
* per disk type volume size is possible in the future. This would require non-trivial code change to handle this in metadata management on master servers.

**Reply by wokalski** [reply]
ah I see. You are very right. Thank you for your answer. I will create a feature request but I understand it's a non trivial change. For now I think I'll run two deployments of seaweedfs 👍
